### PR TITLE
Added support for updated Reserved IP behavior to existing resources (instance, network_ips)

### PR DESCRIPTION
**Pull request by AniJ98**
## 📝 Description

This PR implements functionality and tests for managing Reserved IP addresses in the Linodego client. These changes align with the updated Linode API, providing comprehensive support for Reserved IP operations. The changes include:

Implementation of core operations:

- Deleting a Linode while retaining the Reserved IP
- Retrieving IP information for Linodes, including Reserved IPs
- Listing all IP addresses, including Reserved IPs
- Allocating and assigning new Reserved IP addresses
- Assigning Reserved IPs to Linodes
- Fetching particular IP address, including Reserved IPs
- Converting between Ephemeral and Reserved IPs

Test coverage:

- TestInstance_DeleteInstanceVariants: Verifies Reserved IP retention after Linode deletion
- TestReservedIPAddresses_GetIPReservationStatus: Tests retrieval of Linode IP information including Reserved IPs
- TestIPAddresses_List_smoke: Checks filtering of all IPs by the reserved field.
- TestIPAddress_Instance_Allocate: Covers various scenarios for allocating Reserved IPs
- TestIPAddress_Instance_ReserveIP_Assign: Tests assigning Reserved IPs to Linodes
- TestIPAddress_GetFound: Tests fetching information about a particular IP address, including Reseved IPs
- TestIPAddress_Update: Verifies conversion between Ephemeral and Reserved IPs

## ✔️ How to Test

**What are the steps to reproduce the issue or verify the changes?**

1. Ensure you have a valid Linode API token. Set up your environment: 

`export LINODE_TOKEN="your_token_here"`

   Additionally you need to set the following if you want to test it in a different environment:

```
export LINODE_URL="https://api.linode.com
export LINODE_API_VERSION="v4beta"
```
2. Navigate to the test directory within the linodego project. Update the LINODE_TOKEN in the Makefile:

`LINODE_TOKEN="your_token_here"`

Run the tests using one of the following commands: To run all integration tests:

`make testint`

To run a specific test:

`go test -v ./integration -run TestInstance_DeleteInstanceVariants`

3. Verify the test output for any failures or unexpected behavior.

Note:
Ensure that you have enough quota to reserve IP before adding it to an existing linode. Ensure the IPMAX limit is enough to allocate/ assign a reserved IP to a linode.

### CODE REVIEW

File: test/integration/network_ips_test.go [hunk 1]
```diff
@@ -132,6 +192,184 @@ func TestIPAddress_Update(t *testing.T) {
...
+	_, err = client.UpdateIPAddress(context.Background(), unassignedResIP, updateOpts)
+	if err == nil {
+		t.Fatalf("Expected error when setting RDNS for unassigned reserved IP, but got none")
+	}
+
+	// Sceanrio 6: Try to reserve an IP at MAX reserve IP limit
+
+	reservedIP, err = createReservedIP()
+	if err == nil {
+		t.Fatalf("Expected error indicating MAX IP Reservation limit has been reached, got nil")
```
Comment (lgarber-akamai) [inline test/integration/network_ips_test.go:306]: We typically try to design tests to be account/limit-agnostic so I think we can remove this case

File: test/integration/network_ips_test.go [hunk 2]
```diff
@@ -43,6 +44,33 @@ func TestIPAddress_GetFound_smoke(t *testing.T) {
 	if i.Address != address {
 		t.Errorf("Expected a specific ipaddress, but got a different one %v", i)
 	}
+
+	// Test for fetching reserved IP
+	// First, create a reserved IP
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{
```
Comment (lgarber-akamai) [inline test/integration/network_ips_test.go:50]: Just a heads up, I'm getting 500s when trying to allocate reserved IPs in `ap-west` (where this test's instance is provisioned

File: network_ips.go [hunk 3]
```diff
@@ -88,3 +97,14 @@ func (c *Client) ShareIPAddresses(ctx context.Context, opts IPAddressesShareOpti
 	_, err := doPOSTRequest[InstanceIP](ctx, c, e, opts)
 	return err
 }
+
+// AllocateReserveIP allocates a new IPv4 address to the Account, with the option to reserve it
+// and optionally assign it to a Linode.
+func (c *Client) AllocateReserveIP(ctx context.Context, opts LinodeReserveIPOptions) (*InstanceIP, error) {
+	e := "networking/ips"
```
Comment (lgarber-akamai) [inline network_ips.go:104]: This isn't quite related to this PR but I'm curious about what the difference between `POST networking/reserved/ips` and `POST networking/ips` (with reserved == `true`) is. Is this effectively just an alias?

Comment (AniJ98) [reply network_ips.go:104]: That is a good question! We are currently supporting creation of reserved IP's through both `POST networking/reserved/ips` and `POST networking/ips (with reserved == true)`. The only difference is that along with creation we are also able to assign reserved IP's to linodes using `POST networking/ips (with reserved == true)`.

File: test/integration/network_ips_test.go (hunk 1 above)
Comment (AniJ98) [reply test/integration/network_ips_test.go:306]: Got it! I will go ahead and remove this test.

File: network_ips.go (hunk 3 above)
Comment (lgarber-akamai) [reply network_ips.go:104]: Got it, thanks!

File: network_ips.go [hunk 4]
```diff
@@ -16,6 +17,14 @@ type LinodeIPAssignment struct {
 	LinodeID int    `json:"linode_id"`
 }
 
+type LinodeReserveIPOptions struct {
```
Comment (lgarber-akamai) [inline network_ips.go:20]: nitpick: Could this be renamed to `AllocateReserveIPOptions`?

Comment (AniJ98) [reply network_ips.go:20]: Yes! I will make the change now

**Review by lgarber-akamai — APPROVED** [review]
All tests are passing on my end, great work and thanks for the contribution!

**Review by ezilber-akamai — APPROVED** [review]
Looks good and tests are passing locally. Thanks for the contribution!

**Pull request merged**